pontoon/sync/tests/test_tasks.py

```diff
@@ -161,3 +164,60 @@ def test_remove_duplicate_approvals(self):
         duplicate_translation.refresh_from_db()
         assert_true(duplicate_translation.approved)
         assert_equal(duplicate_translation.approved_date, aware_datetime(1970, 1, 3))
+
+
+class UserError(Exception):
+    pass
+
+
+class SyncExecutionTests(TestCase):
+    def test_serial_task(self):
+        """
+        Test if sync will create lock in cache and release this after task is done.
+        """
+        @serial_task(100)
+        def test_task(self, callback):
+            return callback()
+
+        def execute_second_inner_task():
+            return test_task.delay(lambda: None)
+
+        first_call = test_task.delay(execute_second_inner_task)
+        second_call = first_call.get()
+
+        assert_true(first_call.successful())
+        assert_true(second_call.failed())
+        assert_raises(RuntimeError, second_call.get)
+
+    def test_release_lock_after_timeout(self):
+        """
+        Tests if lock is released after specified timeout.
+        """
+        with patch('pontoon.sync.core.cache') as mock_cache:
+            @serial_task(3)
+            def timeout_task(self):
+                return 42
+            first_call = timeout_task.delay()
+
+            assert_false(cache.get('serial_task.pontoon.sync.tests.test_tasks.timeout_task'))
+            assert_true(first_call.successful())
+            assert_equal(first_call.get(), 42)
+            mock_cache.assert_has_calls([call.add('serial_task.pontoon.sync.tests.test_tasks.timeout_task', True, timeout=3),
```

Comment:
You're asserting too much, here. All we want to know is if the proper timeout was passed. `mock_cache.add.assert_called_with(ANY, ANY, timeout=3)` is plenty.